### PR TITLE
Urgent fix coil type

### PR DESCRIPTION
**Pull request by athoynilimanew**
## Linked Issues

Fixes duplication of CoilType Enum

Closes #{ID}

## Checklist

I confirm that I have completed the following checks:

- [ ] Tests run locally and pass `pytest tests --reactor`
- [ ] Code quality checks run locally and pass `pre-commit run --from-ref develop --to-ref HEAD`
- [ ] Documentation built locally and checked `sphinx-build -W documentation/source documentation/build`

### CODE REVIEW

File: bluemira/equilibria/coils/coil_type.py
Comment (geograham) [inline bluemira/equilibria/coils/coil_type.py:1]: Could you write a simple test for the CoilType class?

File: bluemira/equilibria/coils/_coil.py
Comment (geograham) [inline bluemira/equilibria/coils/_coil.py:1]: This is probably a silly question, but could you explain why CoilType needed to be moved to a separate file? Is it a circular import error related to plotting.py?

**Review by geograham — CHANGES_REQUESTED** [review]

File: bluemira/equilibria/coils/_coil.py
Comment (athoynilimanew) [reply bluemira/equilibria/coils/_coil.py:1]: Exactly. Somehow plotting.py and pf_coil.py were talking to each other and it was the source of circular error

File: bluemira/equilibria/coils/coil_type.py
Comment (athoynilimanew) [reply bluemira/equilibria/coils/coil_type.py:1]: added in the latest commit.

**Review by geograham — APPROVED** [review]

**Review by je-cook — CHANGES_REQUESTED** [review]
This is not ready. I think there were a few things missing from the review of #3021 on top of that. I will start with comments there and come back here.

**Review by je-cook — CHANGES_REQUESTED** [review]
On top of the comments in #3021 I think the coils enum should be moved to base. I think moving it either to constants.py or its own file in base. I'm unsure of the exact location but generally inside the base module

**Comment by athoynilimanew** [comment]
> On top of the comments in #3021 I think the coils enum should be moved to base. I think moving it either to constants.py or its own file in base. I'm unsure of the exact location but generally inside the base module

Moved to constants.py and accordingly changed the imports. moved test function as well

**Comment by athoynilimanew** [comment]
> This is not ready. I think there were a few things missing from the review of #3021 on top of that. I will start with comments there and come back here.

Applied/responded to all required changes (see #3021 for additional comments)

File: bluemira/equilibria/equilibrium.py [hunk 1]
```diff
@@ -1073,13 +1073,8 @@ def set_vcontrol(self, vcontrol_str: Optional[str] = None):
             vcontrol_type = VerticalPositionControlType(vcontrol_str)
             if vcontrol_type is VerticalPositionControlType.VIRTUAL:
                 self.controller = VirtualController(self, gz=2.2)
-            elif vcontrol_type is VerticalPositionControlType.FEEDBACK:
+            if vcontrol_type is VerticalPositionControlType.FEEDBACK:
```
Comment (je-cook) [inline bluemira/equilibria/equilibrium.py:1076]: I think probably more robust is the below as we haven't dealt with any other control type
```suggestion
            else:
```

File: bluemira/geometry/coordinates.py [hunk 2]
```diff
@@ -628,32 +630,29 @@ def rotation_matrix(theta: float, axis: Union[str, np.ndarray] = "z") -> np.ndar
...
             r_matrix = np.array([
                 [1, 0, 0],
                 [0, np.cos(theta), -np.sin(theta)],
                 [0, np.sin(theta), np.cos(theta)],
             ])
-        else:
-            raise CoordinatesError(
-                f"Incorrect rotation axis: {axis}\nplease select from: ['x', 'y', 'z']"
-            )
+
```
Comment (je-cook) [inline bluemira/geometry/coordinates.py:655]:
```suggestion
         else:
             raise NotImplementedError
```

File: bluemira/geometry/tools.py [hunk 3]
```diff
@@ -727,18 +729,13 @@ def convex_hull_wires_2d(
...
-    if hull_plane is HullPlaneType.XZ:
+    hull_plane = HullPlane(plane)
+    if hull_plane is HullPlane.XZ:
         plane_idxs = (0, 2)
-    elif hull_plane is HullPlaneType.XY:
+    elif hull_plane is HullPlane.XY:
         plane_idxs = (0, 1)
-    elif hull_plane is HullPlaneType.YZ:
+    else:  # YZ
         plane_idxs = (1, 2)
```
Comment (je-cook) [inline bluemira/geometry/tools.py:738]:
```suggestion
    elif hull_plane is HullPlaneType.YZ:
        plane_idxs = (1, 2)
    else: 
        raise NotImplementedError
```

**Review by je-cook — APPROVED** [review]

**Pull request merged**